pennylane/vqe/vqe.py

```diff
@@ -41,6 +70,14 @@ class Hamiltonian(qml.operation.Observable):
         observables (Iterable[Observable]): observables in the Hamiltonian expression, of same length as coeffs
         simplify (bool): Specifies whether the Hamiltonian is simplified upon initialization
                          (like-terms are combined). The default value is `False`.
+        compute_grouping (bool): If True, compute and store information on how to group commuting
+            observables upon initialization. This information may be accessed when QNodes containing this
+            Hamiltonian are executed on devices.
+        grouping_type (str): The type of binary relation between Pauli words.
+            Can be ``'qwc'``, ``'commuting'``, or ``'anticommuting'``. Ignored if ``compute_grouping`` is False.
```

Comment:
I wonder if having both of these args is redundant?

Reply:
Good point. Are you thinking
```python
qml.Hamiltonian(obs, coeffs, grouping=None)
qml.Hamiltonian(obs, coeffs, grouping="qwc")
```
etc.

Reply:
yes

Reply:
I would find this a barrier for a user - they have to know something about the grouping to use it. My personal main use case would just be `grouping=True`, I don't want to look up strings of strategies...

Reply:
I admit I like it because:

- it is explicit
- it avoids the value of one argument affecting the behaviour of another (which I think is always confusing!)
- it matches other libraries, e.g., [`scipy.optimize.minimize()`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html)

Reply:
Cool, will change it then. `method` will still be influenced by `grouping_type`, but that's ok I hope...